src/home/templates/partials/_table.html

```diff
@@ -1,420 +1,580 @@
+          );
+          nextButton.textContent = "Next";
+          nextButton.addEventListener("click", () =>
+            fetchAndDisplayReviews(serviceId, current_page + 1)
+          );
+          paginationDiv.appendChild(nextButton);
+        }
 
-            const descriptionElement = document.getElementById('serviceDescription');
-            descriptionElement.innerHTML = '';  // Clear previous content
-
-            const heading = document.createElement('h3');
-            heading.textContent = 'Additional Descriptive Details:';
-            heading.style.marginBottom = '10px';
-            heading.style.fontSize = '1.1em';
-            heading.style.fontWeight = 'bold';
-            descriptionElement.appendChild(heading);
-
-            // Check if Description exists and is an object
-            if (service.Description && typeof service.Description === 'object') {
-                let hasDescription = false;
-                const table = document.createElement('table');
-
-                for (const [key, value] of Object.entries(service.Description)) {
-                    if (value !== null && value !== '') {
-                        hasDescription = true;
-                        const tr = document.createElement('tr');
+        reviewsContainer.appendChild(paginationDiv);
+      } catch (error) {
+        console.error("Error fetching reviews:", error);
+        alert("Failed to load reviews. Please try again.");
+      }
+    }
 
-                        const th = document.createElement('th');
-                        th.textContent = `${key.replace(/_/g, ' ')}:`;
+    // Expose the function globally
+    window.fetchAndDisplayReviews = fetchAndDisplayReviews;
 
-                        const td = document.createElement('td');
-                        td.innerHTML = value.replace(/\n/g, '<br>');
+    async function handleEditReview(review) {
+      try {
+        const newRating = prompt(
+          "Enter a new rating (1-5):",
+          review.RatingStars
+        );
 
-                        tr.appendChild(th);
-                        tr.appendChild(td);
-                        table.appendChild(tr);
-                    }
-                }
+        // Validate the rating input
+        const ratingRegex = /^[1-5]$/; // Only allows values from 1 to 5
 
-                if (hasDescription) {
-                    descriptionElement.appendChild(table);
-                } else {
-                    descriptionElement.textContent = 'No description available.';
-                }
-            } else {
-                descriptionElement.textContent = 'No description available.';
-            }
+        if (!newRating || !ratingRegex.test(newRating)) {
+          alert("Please enter a valid rating between 1 and 5.");
+          return;
+        }
 
-            const getDirectionsBtn = document.getElementById('getDirections');
+        const newMessage = prompt("Enter a new message:", review.RatingMessage);
```

Comment:
same thing here. i think a modal that shows up allowing the user to edit their reviews would look really nice.